### DISCUSSION TITLE
Is it possible to prevent embedding in production builds

### DISCUSSION THREAD

**Discussion by icedice** [bstillitano/scyther#38] [Q&A]
I realise that Scyther.instance.start() only starts the service if it's not a production build, but would it be possible to prevent the library from being embedded in production builds all together?

**Comment by bstillitano** [comment]
This is something that I am interested in exploring however not something I’m aware of SPM offering although I could be wrong on this one.

You can however achieve this using cocoapods if I’m not mistaken, which
scyther supports.

**Reply by icedice** [reply]
Yes, you're right. It's possible with pods but we already use both spm and carthage and i would like to avoid using pods as well.
Anyways I'll see what I can do..  And thanks for an awesome tool.

**Reply by bstillitano** [reply]
No problems, glad you're getting use out of it. 

If you do find a way to do this please let me know and I'll update the documentation to reflect.

**Comment by icedice** [comment]
Btw.. the link to the playground seems to be broken.... I was hoping to learn more about the features (especially how to extend the menu with custom flags and tools...) 

https://github.com/bstillitano/Scyther/tree/improvement/readme/Scyther%20Playground

**Reply by bstillitano** [reply]
Ah yes I've neglected the playground a little. I'll get it updated over the weekend.

**Reply by bstillitano** [reply]
@icedice I've been flat out that last couple weeks but wanted to share the code to add custom tools with you.

```
 // Setup Developer Tools
            var customOption: DeveloperOption = DeveloperOption()
            customOption.name = "Custom Controller"
            customOption.icon = UIImage(systemImage: "link")
            customOption.viewController = UIViewController()
            Scyther.instance.developerOptions.append(customOption)
```

**Reply by icedice** [reply]
> @icedice I've been flat out that last couple weeks but wanted to share the code to add custom tools with you.
> 
> 
> 
> ```
> 
>  // Setup Developer Tools
> 
>             var customOption: DeveloperOption = DeveloperOption()
> 
>             customOption.name = "Custom Controller"
> 
>             customOption.icon = UIImage(systemImage: "link")
> 
>             customOption.viewController = UIViewController()
> 
>             Scyther.instance.developerOptions.append(customOption)
> 
> ```

No worries, thank you for the example that is sufficient for me to move along.
